cpu/samd21/vectors.c

```diff
@@ -22,6 +22,31 @@
 #include <stdint.h>
 #include "vectors_cortexm.h"
 
+#ifdef MODULE_PUF_SRAM
+#include "puf_sram.h"
+
+/* SRAM memory marker defined in the linker script */
+extern uint32_t _srelocate;
+
+bool puf_sram_softreset(void)
+{
+    /*  read "Power On Reset" Bit in RCAUSE register */
+    if(PM->RCAUSE.reg & PM_RCAUSE_POR) {
```

Comment:
Does this filter out hardware reset via reset button/signal and other non software originated resets?

Reply:
From SAMD21 datasheet sect. 7.4.1: *POR monitors VDDANA. It is always activated and monitors voltage at startup and also during all the sleep modes. If VDDANA goes below the threshold voltage, the entire chip is reset.*
The bit is set if a POR occurs.